Corrrupted close sector test

kapi-no opened the pull request:
test to cover yet another corner case

kapi-no committed:
drivers: flash: flash_simulator: added max erase calls param

Added a new runtime parameter to the Flash Simulator. It can be used to
ignore erases to the simulated flash memory after a certain number
of flash erase calls has been executed.

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

kapi-no committed:
drivers: flash: flash_simulator: added max len param

Added a new runtime parameter to the Flash Simulator. It can be used to
ignore part of the data that is written to the simulated flash memory.
This parameter works together with max write calls parameter.

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

kapi-no committed:
tests: fs: nvs: moving nvs_clear to setup method

Flash clearing operation is now moved to the setup method inside the
NVS unit test.

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

kapi-no commented:
cc: @nvlsianpu

nvlsianpu commented on tests/subsys/fs/nvs/src/main.c:
```diff
@@ -244,6 +244,109 @@ void test_nvs_gc(void)
+			*flash_erase_stat = 0;
+
+			/* Block write calls and simulate power down
+			 * during sector closing operation.
```
so only a part of a NVS closing ate will be written.

nvlsianpu commented on tests/subsys/fs/nvs/src/main.c:
```diff
@@ -244,6 +244,109 @@ void test_nvs_gc(void)
+		zassert_mem_equal(buf, rd_buf, sizeof(rd_buf),
+				"RD buff should be equal to the WR buff");
+
+	}
```
can you add one more write and read operation after that to ensure NVS is capable to do so?

kapi-no commented:
@nvlsianpu, I addressed your comments and rebased the PR to deal with merge conflicts.

kapi-no commented:
CI has failed because of some kernel tests. It seems unrelated to this PR, how should I fix it?

nvlsianpu commented:
recheck

kapi-no committed:
tests: fs: nvs: unittest with corrupted sector close op

Added a test case with corner case scenario, in which sector closing
operation is interrupted (e. g. due to power-down).

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

kapi-no commented:
@carlescufi, @nashif could you merge this PR?
